docs/getting-started/builtin-roles/index.rst

```diff
@@ -0,0 +1,52 @@
+==============
+Builtin Roles
+==============
+
+oso includes a library that makes it easy to add roles to your application.
+It integrates with our model library integrations and allows you to create role
+relationships between your user models and your resource models.
+We then generate some polar to make it easy to define rules over these roles instead
+of directly over your users and provide helper methods for managing roles.
+
+Currently this feature is in preview for the SQLAlchemy library.
+
+How it works
+============
+
+Lets say you have a User class and a Widget class and you want to assign users a role for a widget like
+"OWNER" or "User". Using the roles library you can generate a WidgetRole model which allows you to assign
+a user a role for a Widget. The schema for this new model's table looks like this. 
+
+.. image:: /getting-started/builtin-roles/roles.svg
+
+It's a join table between
+User and Widget that contains some metadata like the name of the role. In SQLAlchemy library we add User.widgets
+and Widget.users as relationships you can query, as well as User.widget_roles and Widget.roles to get the roles directly.
```

Comment:
We should include that the convention for naming these is `users` and `Widget.__name__ + "s"`/